### PR TITLE
Auto-populate the NGR value based on address lookup result

### PR DESCRIPTION
**Pull request by autotelik**
WAS-1022 - Auto-populate the NGR value based on address lookup result

populates Location with grid reference details

### CODE REVIEW

File: app/decorators/models/digital_services_core/location_decorator.rb [hunk 1]
```diff
@@ -5,8 +5,28 @@ def any_site_address_attributes_present?
     grid_reference? || site_info?
   end
 
+  after_save :convert_xy_to_grid_ref, if: Proc.new { |l| l.grid_reference.blank? }
```
Comment (rickclare) [inline app/decorators/models/digital_services_core/location_decorator.rb]: Any reason why we can't use this simplified version?

```
after_save :convert_xy_to_grid_ref, unless: grid_reference?
```

File: app/decorators/models/digital_services_core/location_decorator.rb [hunk 2]
```diff
@@ -5,8 +5,28 @@ def any_site_address_attributes_present?
...
+  end
+
+  def self.default_site_info
+    I18n.t("location.default_site_info")
+  end
+
   private
 
+  def convert_xy_to_grid_ref
+    return if(self.x.blank? || self.y.blank?)
```
Comment (rickclare) [inline app/decorators/models/digital_services_core/location_decorator.rb]: Can we simplify the two `return if` lines to:

```
return unless x? && y? && grid_reference.blank?
```

File: app/decorators/models/digital_services_core/location_decorator.rb [hunk 3]
```diff
@@ -5,8 +5,28 @@ def any_site_address_attributes_present?
...
+
+    return unless(self.grid_reference.blank?)
+
+    national_gridref_converter.convert_and_store(self)
+
+    self.update(site_info: DigitalServicesCore::Location.default_site_info) if(self.site_info.blank?)
```
Comment (rickclare) [inline app/decorators/models/digital_services_core/location_decorator.rb]: Might be more readable written as:

```ruby
update(site_info: DigitalServicesCore::Location.default_site_info) unless site_info?
```

i.e. references to `self` is not necessary

File: app/decorators/models/digital_services_core/location_decorator.rb [hunk 4]
```diff
@@ -5,8 +5,28 @@ def any_site_address_attributes_present?
...
+  def self.default_site_info
+    I18n.t("location.default_site_info")
+  end
+
   private
 
+  def convert_xy_to_grid_ref
+    return if(self.x.blank? || self.y.blank?)
+
+    return unless(self.grid_reference.blank?)
```
Comment (rickclare) [inline app/decorators/models/digital_services_core/location_decorator.rb]: Do we need this second `return unless`

**Comment by rickclare** [comment]
Back to @autotelik for review of PR comments

File: app/decorators/models/digital_services_core/location_decorator.rb (hunk 1 above)
Comment (autotelik) [inline app/decorators/models/digital_services_core/location_decorator.rb]: only my ignorance ! so nope. much nicer

File: app/decorators/models/digital_services_core/location_decorator.rb (hunk 2 above)
Comment (autotelik) [inline app/decorators/models/digital_services_core/location_decorator.rb]: agreed

File: app/decorators/models/digital_services_core/location_decorator.rb (hunk 3 above)
Comment (autotelik) [inline app/decorators/models/digital_services_core/location_decorator.rb]: agreed

File: app/decorators/models/digital_services_core/location_decorator.rb (hunk 4 above)
Comment (autotelik) [inline app/decorators/models/digital_services_core/location_decorator.rb]: was just belts and braces for potential future  calls to convert_xy_to_grid_ref

**Comment by autotelik** [comment]
@rickclare think all suggestions incorporated

**Pull request merged**